### PR TITLE
MULE-19928: initialize object store inside redelivery policy

### PR DESCRIPTION
**Pull request by jcgrethe**

### CODE REVIEW

**Comment by jcgrethe** [comment]
--validate

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ComponentModel.java [hunk 1]
```diff
@@ -411,6 +412,28 @@ private void handleNestedParameters(ComponentModel componentModel, Stream<Compon
                                       Multimap<ComponentIdentifier, ComponentModel> nestedComponents,
                                       ExtensionModelHelper extensionModelHelper,
                                       ParameterizedModel model, Predicate<ParameterModel> parameterModelFilter) {
+
+    componentModel.getParent().getInnerComponents().stream()
```
Comment (elrodro83) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ComponentModel.java:416]: add a comment explaining the situation why thos code is needed and what it does

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ComponentModel.java [hunk 2]
```diff
@@ -411,6 +412,28 @@ private void handleNestedParameters(ComponentModel componentModel, Stream<Compon
...
+        .filter(innerComponentModel -> innerComponentModel.getIdentifier().getName().equals("redelivery-policy")).findAny()
```
Comment (elrodro83) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ComponentModel.java:417]: reuse an already exsiting constant

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ComponentModel.java [hunk 3]
```diff
@@ -411,6 +412,28 @@ private void handleNestedParameters(ComponentModel componentModel, Stream<Compon
...
+        .ifPresent(redeliveryComponentModel -> {
+          model.getAllParameterModels().stream()
+              .filter(parameterModel -> parameterModel.getName().equals("redeliveryPolicy")).findAny()
```
Comment (elrodro83) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ComponentModel.java:420]: constant

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ComponentModel.java (hunk 2 above)
Comment (elrodro83) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ComponentModel.java:417]: be careful, you are ignoring the namespace of the component here

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ComponentModel.java [hunk 4]
```diff
@@ -411,6 +414,30 @@ private void handleNestedParameters(ComponentModel componentModel, Stream<Compon
                                       Multimap<ComponentIdentifier, ComponentModel> nestedComponents,
                                       ExtensionModelHelper extensionModelHelper,
                                       ParameterizedModel model, Predicate<ParameterModel> parameterModelFilter) {
+
+    //MULE-19928: munit tests fail to initialize an object store inside a redelivery policy.
```
Comment (elrodro83) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ComponentModel.java:418]: explain the fix, not the symptom of the issue

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ComponentModel.java [hunk 5]
```diff
@@ -411,6 +414,30 @@ private void handleNestedParameters(ComponentModel componentModel, Stream<Compon
...
+    componentModel.getParent().getInnerComponents().stream()
+        .filter(innerComponentModel -> innerComponentModel.getIdentifier().getName().equals(REDELIVERY_POLICY_ELEMENT_IDENTIFIER)
```
Comment (elrodro83) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/model/ComponentModel.java:420]: still using just the name here, use the full identifier for equality check

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**